pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -170,6 +160,10 @@ if [ "$dontLink" != 1 ]; then
     export NIX_LINK_TYPE_@suffixSalt@=$linkType
 fi
 
+if [ "$setClangTarget" == 1 ]; then
+    extraAfter+=(-target $clangTarget)
```

Comment:
@Ericson2314 will empty @ variables also be substituted with empty string/nothing or do they get skipped over and left as is?

We still need some way to pass `setClangTarget` from default.nix into cc-wrapper.sh. I can either do something like `export setClangTarget=1` and then instead of of `if [ $setClangTarget...` I can do `if [ @isClang@`. But if unknown @ vars are left as empty then I can do something like:

```
clangTarget=@clanTarget@

...

case $p in
...
-target) clangTarget=$p2
...
esac

...

if [ -n $clangTarget ]; then
    extraAfter+=(-target $clangTarget)
fi
```

Reply:
I would call `setClangTarget` `needsTarget` and just it `0` or `1` based on the whether is clang in default.nix

Let's try to make sure all variables to substitute are defined to *something* at build time,

Reply:
:+1:, also just read from https://nixos.org/manual/nixpkgs/stable/#fun-substituteAll that unknown @ vars are left as is so wouldn't work the way I laid it out in the snippet above.

Reply:
Should be good now.